blog/2025/passkeys-support-26-4.adoc

```diff
@@ -0,0 +1,105 @@
+:title: Passkeys support in upcoming Keycloak release (26.4)
+:date: 2025-09-16
+:publish: true
+:author: Peter Skopek
+:summary: Keycloak 26.4 will bring passkeys as supported feature. There are seamlessly integrated to our build in browser flow and all forms containing username or password fields.
+
+Passkeys have been available in Keycloak since version 23.0.0 as a preview feature. We are happy to announce official support for passkeys in upcoming **Keycloak 26.4.0**.
+
+== What is passkey?
+
+Definition from https://fidoalliance.org[FIDO Alliance]
+====
+A passkey is a FIDO authentication credential based on FIDO standards, that allows a user to sign in to apps and websites with the same process that they use to unlock their device (biometrics, PIN, or pattern).
+Passkeys are FIDO cryptographic credentials that are tied to a user’s account on a website or application.
+====
+
+https://www.passkeycentral.org/home[More info]
+
+== What's new?
+
+Passkeys are now seamlessly integrated to Keycloak using both conditional and modal UI. There is no need to modify default browser flow to use passkeys. Passkeys support is not enabled aby default. It needs to be enabled in the *WebAuthn Passwordless Policy* (*Authentication* → *Policies* → *Webauthn Passwordless Policy*).
+
+There is new *Conditional - credential* authenticator that checks if a specific credential type (passkey) has been used during
+the authentication process. It is added to the default *browser* flow to skip 2FA in case a passkey was used to log in as
+the primary credential.
+
+Further quite hidden passkey support is also in the re-authentication form, where users can choose passkey as well as password.
+
+For more information check https://www.keycloak.org/docs/latest/server_admin/index.html#_passkeys-conditional-ui[Keycloak Server Administration Guide].
+
+== A few examples
+
+Let's start form the fresh Keycloak database. The first steps are obvious ones.
+
+- create admin user
+- create one test user
+- enable passkey support in (*Authentication* → *Policies* → *Webauthn Passwordless Policy*)
+- set default required action in (*Authentication* → *Required actions* → *Webauthn Register Passwordless*)
+
+No need for any other changes.
+
+=== Conditional UI
+
+Conditional UI is displayed when other components of the platform supports this UI stile.
```

Comment:
Maybe we can add that the conditional UI is always presented when a username input is used. I mean, in all default forms that require the username for login.